modules/validations/duplicate_address.js

```diff
@@ -0,0 +1,136 @@
+import { t } from '../core/localizer';
+import { utilArrayIntersection, utilDisplayLabel } from '../util';
+import { validationIssue } from '../core/validation';
+
+
+export function validationDuplicateAddress(context) {
+  const type = 'duplicate_address';
+
+
+  function hasAddress(entity) {
+    return Object.keys(entity.tags).some(k => /^addr:/.test(k));
+  }
+
+
+  function isDuplicate(entity1, entity2) {
+    if (entity1.id === entity2.id) return false;   // ignore self
+
+    let dupe = { entity1: entity1, entity2: entity2, tags: {} };
+    let different = false;
+
+    // if any of these tags don't match, *and both are present*, it's not a duplicate..
+    ['addr:street', 'addr:place', 'addr:block'].forEach(k => {
+      const val1 = entity1.tags[k] || '';
+      const val2 = entity2.tags[k] || '';
+      if (val1 && val2) {     // both assigned
+        if (val1 !== val2) {
+          different = true;   // values don't match
+        } else {
+          dupe.tags[k] = val1;
+        }
+      }
+    });
+    if (different) return false;
+
+    // if any of these tags don't match, *and one is present*, it's not a duplicate..
+    ['addr:door', 'addr:unit', 'addr:flats', 'addr:floor'].forEach(k => {
```

Comment:
Shouldn't you swap this list of keys with the list in line 22 (`['addr:street', 'addr:place', 'addr:block']`)? Otherwise two objects with identical house number but one with `addr:street=ABC Street` and the other one with `addr:place=DEF Farm` are not detected as different.